project0/index.js

```diff
@@ -0,0 +1,283 @@
+// implementation was based on the MVC pattern
+
+class QueueModel {
+  constructor() {
+    this.patients = [];
+    this.currentPatientId = undefined;
+    this.currentPatient = undefined;
+    this.currentPatientName = undefined;
+    this.searchedResolution = '';
+  }
+
+  addPatient(patientName) {
+    console.log(this.patients);
+
+    const duplicatePatient = this.patients.some((patient) => patient.name === patientName.toUpperCase());
+
+    if (!duplicatePatient) {
+      const patient = {
+        id: this.patients.length > 0 ? this.patients[this.patients.length - 1].id + 1 : 1,
+        name: patientName.toUpperCase(),
+        resolution: undefined,
+      };
+
+      this.patients.push(patient);
+    } else {
+      throw Error('Database already contains this patient!');
```

Comment:
Better move all messages to separate file and constants `messages.js`

Reply:
I strongly recommend using CustomError object for error handling. 
More about: 
https://levelup.gitconnected.com/node-js-best-practices-error-handling-50719c42504f
https://www.smashingmagazine.com/2020/08/error-handling-nodejs-error-classes/#error-classes